### DISCUSSION TITLE
Two missing features maybe

### DISCUSSION THREAD

**Discussion by ASCI-Blue** [streamdeck-linux-gui/streamdeck-linux-gui#90] [Ideas / Feature Requests]
Started using Streamdeck UI a short while ago and it doesn't recognize arrow key inputs. Not sure if this is Manjaro, Stream Deck UI, or my keyboard (Sol 3 running QMK) being a pain. 

Also any intent to add a wallpaper feature for when Streamdeck goes to sleep?

**Comment by Danie10** [comment]
Not sure if you meant output as arrow keys? If so you can use xdotool command, or as Press Keys you can just enter Up or Right etc.

**Comment by ASCI-Blue** [comment]
I think I wanted to populate 'command.' I'm use my Stream Deck as a partial controller for Starfield. Part of the flight model has 6 systems (left and right) and power increase/decrease (up/down). I added up-down-left-right to individual buttons in Press Keys and it works.

**Comment by Danie10** [comment]
Great news! One thing that I opened an issue for still, is repeat key while being button being held down. It does not yet recognise/react on holding a button, which is needed for many games.

**Comment by ASCI-Blue** [comment]
Quickie bump to this, the software seems to work except in Starfield. I did a pretty extensive system update today which broke most of what I'd done, updated to 3.2 beta, edited the command to start the game. Nothing in SF worked so not sure if it's a Proton matter or something else. I tried Cyberpunk and the few crossover key commands work in it just fine.

**Comment by ASCI-Blue** [comment]
Additional bump, experimented across 3.10, 3.20. All games played via Steam. up-down-left-right work sometimes in Cyberpunk 2077 but only on the esc menu along with wasd. M works in game to bring up the map. BG3 L and M both do their jobs, again WASD/Up-down-left-right don't function. X4 Foundations, linux native wasd/up-down-left-right don't function as the character but work on the map.

**Comment by Data-007** [comment]
@ASCI-Blue X11 or Wayland? More info please.
Hit us up on Discord. https://discord.gg/BRPBSbvaD4

**Comment by Data-007** [comment]
So you are on X11 thanks for the information

**Comment by phptek** [comment]
Also keen on the wallpaper feature. I bought my SD off a friend who used MacOS. I had to get him over to use his MacOS StreamDeck app to change the wallpaper on my StreamDeck. Would be cool to have this feature under the main settings button.

Ubuntu 22.04 / X11

**Reply by Danie10** [reply]
@phptek maybe you want to consider migrating to Stream Controller like many of us have already done, as it has wallpaper as well as add-on modules for this. It's doing all the things that I'd wanted before, like repeat whilst holding etc - https://core447.com/

**Reply by coolapso** [reply]
@phptek  providing more insights on the @Danie10  answer: https://github.com/streamdeck-linux-gui/streamdeck-linux-gui/discussions/203

**Reply by phptek** [reply]
Which release of StreamController are folks using? I'm on Ubuntu 22.04 but cannot install it (`main` and `1.4.9-beta`) due to some meson deps missing.

**Reply by coolapso** [reply]
Unless you want to do any development, use the flatpak instead.

**Reply by Danie10** [reply]
I'm using the flatpak and it is working fine.

**Reply by coolapso** [reply]
anyway here are the installation instructions for both flatpak and python: https://streamcontroller.github.io/docs/latest/installation/

**Reply by phptek** [reply]
Cheers all! StreamController is now where it's at. USB connectivity is a little sketchy and it needs to be plugged/unplugged every so often - I put that down to going through my Anker dock. Thanks again!
